pkg/apis/apis/v1alpha1/types_apiexport.go

```diff
@@ -196,6 +201,19 @@ type PermissionClaim struct {
 	IdentityHash string `json:"identityHash,omitempty"`
 }
 
+type ClaimedResource struct {
+	// name of an object within a claimed group/resource.
+	// It matches the metadata.name field of the underlying object.
+	// If unset, all objects having the group/resource are being claimed.
+	// +optional
+	Name string `json:"name,omitempty"`
```

Comment:
Can we (trivially) add validation for these fields to match what an object's name would have to be?

Reply:
sorry i am not following 🙈 can you elaborate? do we have restrictions on names?

Reply:
I think we'd have to ensure we match one or more of these restrictions: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names

tl;dr - if we want to put the names in labels or paths, we'll need to make sure that the name provided follows those constraints.

Reply:
hrm, i believe this would indeed need to be checked during admission. let me see if this can be done declaratively.

Reply:
If we can't trivially do it via some kube-builder annotation or something for validation, maybe not worth it.

Reply:
The closest I can see with the kube-builder annotations (defined [here](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.0.md#schemaObject) would be a combo on maxLength and a regex pattern.

[API Machinery has some validation helpers](https://github.com/kubernetes/kubernetes/blob/c18c2447cf602ee1a76113200b3f5b6200a24a34/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go) that appear to be used within Kube itself.

Reply:
updated with the referenced regex but fyi: I just found a second ago a precedence for this in our code 😛 https://github.com/kcp-dev/kcp/blob/main/pkg/apis/apis/v1alpha1/types_apiexport.go#L229